packages/ontario-design-system-component-library/src/components/ontario-task/ontario-task.tsx

```diff
@@ -0,0 +1,230 @@
+			.addDesignSystemTag()
+			.addMonospaceText(' taskStatus ')
+			.addRegularText('on')
+			.addMonospaceText(' <ontario-task> ')
+			.addRegularText('was set to an invalid taskStatus; only ')
+			.addMonospaceText(TaskStatuses.join(', '))
+			.addRegularText(' are supported. The default taskStatus ')
+			.addMonospaceText('notStarted')
+			.addRegularText(' is assumed.')
+			.printMessage();
+		return 'notStarted';
+	}
+
+	/**
+	 * Watch for changes in `hintText` prop and parse it if available.
+	 */
+	@Watch('hintText')
+	private parseHintText() {
+		if (this.hintText) {
+			const hintTextObject = constructHintTextObject(this.hintText);
+			this.internalHintText = hintTextObject;
+		}
+	}
+
+	/**
+	 * Listen for app language settings on the window to update the component language.
+	 */
+	@Listen('setAppLanguage', { target: 'window' })
+	handleSetAppLanguage(event: CustomEvent<Language>) {
+		if (!this.language) {
+			this.language = validateLanguage(event);
+		}
+	}
+
+	/**
+	 * Listen for header language toggling events to update the language dynamically.
+	 */
+	@Listen('headerLanguageToggled', { target: 'window' })
+	handleHeaderLanguageToggled(event: CustomEvent<Language>) {
+		this.language = validateLanguage(event);
+	}
+
+	/**
+	 * Determines the badge colour based on the current `taskStatusState`.
+	 *
+	 * @returns {TaskBadgeColour}
+	 */
+	private getBadgeColour(): TaskBadgeColour {
+		return TaskToBadgeColour[this.taskStatusState] || 'grey';
+	}
+
+	/**
+	 * Provides a translated task status string based on the current language.
+	 */
+	private getTranslatedTaskStatus(): string {
+		const resolvedLanguage = validateLanguage(this.language);
+		return translations.taskStatus[this.taskStatusState][resolvedLanguage];
+	}
+
+	/**
+	 * Renders hint text if available.
+	 */
+	private renderHintText() {
+		if (this.internalHintText) {
+			return (
+				<ontario-hint-text
+					hint={this.internalHintText.hint}
+					hintContentType={this.internalHintText.hintContentType}
+					ref={(el) => (this.hintTextRef = el)}
+				></ontario-hint-text>
+			);
+		}
+		return null;
+	}
+
+	private getClass(): string {
+		return this.hintText ? `ontario-task__label ontario-task__hint-text--true` : `ontario-task__label`;
+	}
+
+	/**
+	 * Renders the task label and status content.
+	 */
+	private renderTaskContent() {
+		return (
+			<div class="ontario-task__content">
+				<h3 id="task-label" class={this.getClass()}>
+					{this.label}
+				</h3>
+				{this.taskStatusState && (
+					<ontario-badge
+						class="ontario-task__badge"
+						role="status"
+						aria-label={`${
+							translations.taskStatus.taskStatus[validateLanguage(this.language)]
+						} ${this.getTranslatedTaskStatus()} (${this.taskStatusState})`}
+						colour={this.getBadgeColour()}
+					>
+						{this.getTranslatedTaskStatus()}
+					</ontario-badge>
+				)}
+			</div>
+		);
+	}
+
+	async componentDidLoad() {
+		this.hintTextId = await this.hintTextRef?.getHintTextId();
+	}
+
```

Comment:
This could lead to a race condition. This should be set synchronously during `componentWillLoad` if that is possible. Otherwise, if it must stay async, we should provide a fallback or something.